src/server/services/wfs/qgswfsdescribefeaturetype.cpp

```diff
@@ -286,12 +286,18 @@ namespace QgsWfs
       const QgsEditorWidgetSetup setup = field.editorWidgetSetup();
       if ( setup.type() ==  QStringLiteral( "DateTime" ) )
       {
+        // Get editor widget setup config
         const QVariantMap config = setup.config();
-        const QString fieldFormat = config.value( QStringLiteral( "field_format" ), QgsDateTimeFieldFormatter::defaultFormat( field.type() ) ).toString();
-        if ( fieldFormat == QLatin1String( "yyyy-MM-dd" ) )
-          attElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "date" ) );
-        else if ( fieldFormat == QLatin1String( "HH:mm:ss" ) )
+        // Get field format from editor widget setup config
+        const QString fieldFormat = config.value(
+                                      QStringLiteral( "field_format" ),
+                                      QgsDateTimeFieldFormatter::defaultFormat( field.type() )
+                                    ).toString();
+        // Define type from field format
+        if ( fieldFormat == QgsDateTimeFieldFormatter::TIME_FORMAT ) // const TIME_FORMAT
           attElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "time" ) );
+        else if ( fieldFormat == QLatin1String( "yyyy-MM-dd" ) ) // QgsDateTimeFieldFormatter provide a local date format
```

Comment:
```suggestion
        else if ( fieldFormat == QgsDateTimeFieldFormatter::DATE_FORMAT ) 
```

Reply:
I gave up the idea of using `QgsDateTimeFieldFormatter::DATE_FORMAT` because it is not a constant variable. It can be updated by localization and will not be always the same as `"yyyy-MM-dd"`, and `date` type described in an XSD has to be formatted to `yyyy-MM-dd`.

Reply:
Ok thanks for the explanation